include/nlohmann/json.hpp

```diff
@@ -3447,7 +3447,8 @@ class basic_json // NOLINT(cppcoreguidelines-special-member-functions,hicpp-spec
         // swap only works for arrays
         if (JSON_HEDLEY_LIKELY(is_array()))
         {
-            std::swap(*(m_value.array), other);
+            using std::swap;
+            swap(*(m_value.array), other);
```

Comment:
While you're here, should the error strings be updated to provide the parameter type too?

    "cannot use swap(array_t&) with "

Reply:
Good idea. Done.

Reply:
I need to change the unit tests as well. Duh!